status/callrpc.nim

```diff
@@ -52,21 +52,28 @@ type RemoteMethod* {.pure.} = enum
   net_listening = "net_listening"
 
 
-proc newWeb3*(settings: Settings): Web3 {.raises: [CatchableError, Defect,
-  ref Web3Error].} =
-
-  let network = settings.getCurrentNetwork()
+proc getWeb3Conn(networkName: string, network: Option[Network]): Web3 {.raises: 
```

Comment:
This is done this way so I can pass the result of `settings.getNetwork(networkName)` and `settings.getCurrentNetwork()` directly to the `getWeb3Conn` and have this proc check if the network contains a value,  opossed to having this verification done in both `newWeb3` procs. 
Also `getWeb3Conn` is not a procedure that can be exported. it's a helper for `newWeb3`. Perhaps it should have a different name to avoid this confusion?

Reply:
Ah I see, I had forgotten that `getNetwork` and `getCurrentNetwork` returned `Option[Network]` 👍